pkg/operator/ceph/object/dependents.go

```diff
@@ -43,6 +46,24 @@ func CephObjectStoreDependents(
 	baseErrMsg := fmt.Sprintf("failed to get dependents of CephObjectStore %q", nsName)
 
 	deps := dependents.NewDependentList()
+	if store.Spec.IsMultisite() {
```

Comment:
The flow of this will be easier to follow when reading by leaning more heavily on fail-fast methods. Code that is easy to read is easy to maintain, and I following the logic of this as it is structured now is challenging.

```go
if store.Spec.IsMultisite() {
  # stores in multisite configs have different conditions that change what dependents should be checked
  zoneIsMaster, err := CheckZoneIsMaster(objCtx)
  if err != nil {
    return deps, errors.Wrapf(err, baseErrMsg)
  }
  if ! zoneIsMaster {
    # zone is a peer, and the master is the source of truth
    # we assume master is up-to-date and this peer can be deleted safely
    return deps, nil
  }
  err := getMasterZoneDependents(deps, clusterdCtx, clusterInfo, store, objCtx)
  if err != nil {
    return deps, errors.Wrapf(err, /* good err msg */)
  }
  if !deps.Empty() {
    return deps, errors.Errorf(/* super helpful error message */)
  }
  # if we reached this point, this is a master zone with NO peers. it is the last store
  # that has this data, so we should prevent users from accidentally deleting it by
  # continuing to check for buckets
}
```